pkg/build/gobuild.go

```diff
@@ -355,3 +362,130 @@ func (gb *gobuild) Build(s string) (v1.Image, error) {
 	}
 	return image, nil
 }
+
+// Build implements build.Interface
+func (gb *gobuild) Build(s string) (steve.Interface, error) {
+	// Determine the appropriate base image for this import path.
+	base, err := gb.getBase(s)
+	if err != nil {
+		return nil, err
+	}
+
+	switch base.Type() {
+	case types.OCIImageIndex, types.DockerManifestList:
+		idx, err := base.ImageIndex()
+		if err != nil {
+			return nil, err
+		}
+		built, err := newIndex(idx)
+		if err != nil {
+			return nil, err
+		}
+		im, err := idx.IndexManifest()
+		if err != nil {
+			return nil, err
+		}
+		for _, desc := range im.Manifests {
+			baseImage, err := idx.Image(desc.Digest)
+			if err != nil {
+				return nil, err
+			}
+			img, err := gb.buildImage(s, baseImage)
+			if err != nil {
+				return nil, err
+			}
+			if err := built.AddImage(img, desc); err != nil {
+				return nil, err
+			}
+		}
+		return steve.Index(built)
+	default:
+		baseImage, err := base.Image()
+		if err != nil {
+			return nil, err
+		}
+		img, err := gb.buildImage(s, baseImage)
+		if err != nil {
+			return nil, err
+		}
+		return steve.Image(img)
+	}
+
+	return nil, fmt.Errorf("oops")
+}
+
+type index struct {
```

Comment:
All of this could go away with https://github.com/google/go-containerregistry/issues/474